index.js

```diff
@@ -141,44 +175,59 @@ const line = []
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+let i = 0
+function takeATicketNumber(line) {
+  i ++
+  line.push(i)
+  return `Welcome. You are ticket number ${i}`
+}
+console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(nowServing(line))
+// => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+function ticketNumberGeneratorFunc() {
+  let i = 0
+function takeATicketNumber(line) {
+  i ++
+  line.push(i)
+  return `Welcome. You are ticket number ${i}`
+}
+  return takeATicketNumber
+}
```

Comment:
Be mindful of the indentation. Also you are defining a function within another function. This can be achieved with an anonymous function.

```
function ticketNumberGeneratorFunc() {
  let number = 0;
  return (function (line) {
    number += 1;
    line.push(number);
    return `Welcome. You are ticket number ${number}`;
  })
}
```